pytorch_lightning/core/lightning.py

```diff
@@ -405,8 +405,18 @@ def log(
         on_epoch = self.__auto_choose_log_on_epoch(on_epoch)
 
         results = self.trainer._results
-        assert results is not None
-        assert self._current_fx_name is not None
+        if results is None:
+            caller = inspect.stack()[1][3]
```

Comment:
based on my simple testing here I believe this will not return the right answer if self.log is called in a function nested under the `current_fx_name` function. Will this not be misleading?

Reply:
does not matter

Reply:
Actually, let me undo this because the exception prints the stacktrace already

I believe I did this because it made it easier to debug issues during the PR's implenentation